lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/dataset/ReferenceDatasetUtils.scala

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.lightgbm.dataset
+
+import com.microsoft.azure.synapse.ml.lightgbm.swig.SwigUtils
+import com.microsoft.azure.synapse.ml.lightgbm._
+import com.microsoft.ml.lightgbm._
+import org.apache.spark.sql._
+import org.slf4j.Logger
+
+
+object ReferenceDatasetUtils {
+  def createReferenceDatasetFromSample(datasetParams: String,
+                                       featuresCol: String,
+                                       numRows: Long,
+                                       numCols: Int,
+                                       sampledRowData: Array[Row],
+                                       measures: InstrumentationMeasures,
+                                       log: Logger): Array[Byte] = {
+    log.info(s"For reference Dataset creation, received sample data with ${sampledRowData.length} samples")
+
+    // Pre-create allocated native pointers so it's easy to clean them up
+    val datasetVoidPtr = lightgbmlib.voidpp_handle()
+    val lenPtr = lightgbmlib.new_intp()
+    val bufferHandlePtr = lightgbmlib.voidpp_handle()
+
+    val sampledData: SampledData = new SampledData(sampledRowData.length, numCols)
+    try {
+      // create properly formatted sampled data
+      measures.markSamplingStart()
+      sampledRowData.zipWithIndex.foreach(rowAndIndex =>
+        sampledData.pushRow(rowAndIndex._1, rowAndIndex._2, featuresCol))
+      measures.markSamplingStop()
+
+      // Create dataset from samples
+      log.info(s"Creating reference training dataset, config:$datasetParams")
```

Comment:
Make info message prefix consistent. See the one above.

I'd go with this one, e.g. `Creating reference training dataset`.